network/eni/eni.go

```diff
@@ -102,6 +97,27 @@ func (eni *ENI) AttachToLink() error {
 	return nil
 }
 
+// getInterfaceByMac gets the interface with specified mac address, and picks the one with shortest name
+// if there are multiple matches, which can happen temporarily when setting up a branch ENI, in which it
+// can have the same mac address as the trunk during branch.AttachToLink.
+func getInterfaceByMac(macAddress net.HardwareAddr, interfaces []net.Interface) *net.Interface {
+	var chosenInterface net.Interface
```

Comment:
If you define this as a pointer (var chosen *net.Interface), then

the if check at line 108 can become simply
if chosen == nil || len(chosen.Name) > len(i.Name)

line 109 becomes a pointer copy instead of a struct copy

you can remove the if block lines 114-116.

Reply:
i originally did that. then i found the variable "i" in the for loop shares the same address, so i can't just save the address.

Reply:
discussed offline. fixed.